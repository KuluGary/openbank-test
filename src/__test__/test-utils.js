import React from "react"
import { render as rtlRender } from "@testing-library/react"
import { Provider } from "react-redux"

import reduxStore from "store";

/**
 * Wraps the component in a mock of the redux store provider
 * @param {Component} ui - the UI component passed on render
 * @param {Object} renderOptions - the render options passed 
 */
const render = (ui, { initialState, store = reduxStore, ...renderOptions } = {}) => {
    const Wrapper = ({ children }) => {
        return <Provider store={store}>{children}</Provider>
    }

    return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from "@testing-library/react"; // export full testing-library components
export { render }; // export provider wrapped render
