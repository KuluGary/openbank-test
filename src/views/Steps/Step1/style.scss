.step1 {
    padding: 1.5rem 3rem;
    
    &-wrapper {
        display: flex;
        justify-content: space-around;
        flex-wrap: wrap;

        @media (max-width: 768px) {
            display: block;
            margin: 0 auto;
        }

        &-image-container {
            max-width: 35%;
            text-align: justify;

            @media (max-width: 768px) {
                max-width: 75%;
                margin: 0 auto;
            }

            img {
                display: block;
                margin: 0 auto;
            }
        }
    }
}