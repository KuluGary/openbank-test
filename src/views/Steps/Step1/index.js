import React from "react";
import ProtectPassword from "assets/img/group.svg";
import MasterKey from "assets/img/group-3.svg";
import { useTranslation } from "react-i18next";
import "./style.scss";
import { Checkbox, FormControlLabel } from "@material-ui/core";

/**
 * Function component for rendering the first step
 * of the wizard
 */
const Step1 = ({ validator, setValidator }) => {
  const { t } = useTranslation();
  
  // Set default state of current validator to false
  React.useEffect(() => setValidator(false), [setValidator]);

  const handleCheckChange = () => {
    setValidator(!validator);
  };

  return (
    <div className="step1" data-testid="step1">
      <h1>{t("wizard.header")}</h1>
      <div className="step1-wrapper">
        <div className="step1-wrapper-image-container">
          <img alt="Protect your password" src={ProtectPassword} />
          <p>{t("wizard.step1.protect_password")}</p>
        </div>
        <div className="step1-wrapper-image-container">
          <img alt="Protect your password" src={MasterKey} />
          <p>{t("wizard.step1.master_key")}</p>
        </div>
      </div>
      <h4>{t("wizard.step1.how_it_works_header")}</h4>
      <p>{t("wizard.step1.how_it_works")}</p>

      <h4>{t("wizard.step1.which_data_header")}</h4>
      <p>{t("wizard.step1.which_data")}</p>
      <div className="step1-checkbox">
        <FormControlLabel
          control={
            <Checkbox
              checked={validator}
              onChange={handleCheckChange}
              inputProps={{
                "data-testid": "privacy-checkbox",
              }}
            />
          }
          label={t("wizard.step1.checkbox")}
        />
      </div>
    </div>
  );
};

export default Step1;
