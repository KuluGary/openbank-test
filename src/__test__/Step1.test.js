import React from 'react'
import Step1 from 'views/Steps/Step1'

import { render, cleanup } from './test-utils'
import '@testing-library/jest-dom/extend-expect'

const props = {
  validator: false,
  setValidator: () => {}
}

describe('Step 1', () => {
  afterEach(cleanup)

  it('matches snapshot', () => {
    const div = document.createElement('div')
    const component = render(<Step1 {...props} />, div)
    expect(component.container).toMatchSnapshot()
  })

  it('renders component', () => {
    const { getByTestId } = render(<Step1 {...props} />)

    expect(getByTestId('step1')).toBeInTheDocument()
  })
})
