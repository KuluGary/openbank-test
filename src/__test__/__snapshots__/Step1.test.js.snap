// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step 1 matches snapshot 1`] = `
<div>
  <div
    class="step1"
    data-testid="step1"
  >
    <h1>
      wizard.header
    </h1>
    <div
      class="step1-wrapper"
    >
      <div
        class="step1-wrapper-image-container"
      >
        <img
          alt="Protect your password"
          src="group.svg"
        />
        <p>
          wizard.step1.protect_password
        </p>
      </div>
      <div
        class="step1-wrapper-image-container"
      >
        <img
          alt="Protect your password"
          src="group-3.svg"
        />
        <p>
          wizard.step1.master_key
        </p>
      </div>
    </div>
    <h4>
      wizard.step1.how_it_works_header
    </h4>
    <p>
      wizard.step1.how_it_works
    </p>
    <h4>
      wizard.step1.which_data_header
    </h4>
    <p>
      wizard.step1.which_data
    </p>
    <div
      class="step1-checkbox"
    >
      <label
        class="MuiFormControlLabel-root"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              class="PrivateSwitchBase-input-4"
              data-indeterminate="false"
              data-testid="privacy-checkbox"
              type="checkbox"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          wizard.step1.checkbox
        </span>
      </label>
    </div>
  </div>
</div>
`;
